### PR TITLE
Partition plan adaptation to database management.

### PR DESCRIPTION
**Pull request by guowl3**
#### What type of PR is this?
type-feature

#### What this PR does / why we need it:

Partition plan adaptation to database management.

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-migrate/src/main/resources/migrate/common/V_4_2_0_42__alter_partition_plan.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+ALTER TABLE `table_partition_plan` MODIFY COLUMN `flow_instance_id` bigint(20) COMMENT '任务 ID' NULL;
```
Comment (yizhouxw) [inline server/odc-migrate/src/main/resources/migrate/common/V_4_2_0_42__alter_partition_plan.sql:1]: if a column should not be null, use `NOT NULL`

File: server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/PartitionPlanController.java [hunk 2]
```diff
@@ -28,38 +27,36 @@
...
 @RestController
-@ConditionalOnProperty(name = "odc.feature.partitionplan.enabled", havingValue = "true")
 public class PartitionPlanController {
 
     @Autowired
     private PartitionPlanService partitionPlanService;
 
-    @RequestMapping(value = "/api/v2/patitionplan/ConnectionPartitionPlan", method = RequestMethod.GET)
-    public SuccessResponse<ConnectionPartitionPlan> getConnectionPartitionPlan(@RequestParam Long connectionId,
+    @RequestMapping(value = "/api/v2/patitionplan/DatabasePartitionPlan/", method = RequestMethod.GET)
```
Comment (yizhouxw) [inline server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/PartitionPlanController.java:43]: seems no RESTful, what is the relationship between database and partition plan ?

may `api/v2/partitionplan/PartitionPlans/{id}` ?

File: server/odc-migrate/src/main/resources/migrate/common/V_4_2_0_42__alter_partition_plan.sql (hunk 1 above)
Comment (guowl3) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_2_0_42__alter_partition_plan.sql]: done

File: server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/PartitionPlanController.java (hunk 2 above)
Comment (guowl3) [reply server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/PartitionPlanController.java]: The API has been modified to 'api/v2/partitionplan/partitionPlans'

**Review by krihy — APPROVED** [review]
LGTM

**Pull request merged**